os-server/src/main/java/app/openschool/user/UserController.java

```diff
@@ -104,16 +106,37 @@ public ResponseEntity<CourseDto> deleteCourse(
   }
 
   @PostMapping("/courses/{courseId}")
-  @Operation(summary = "enroll course")
+  @Operation(summary = "enroll a course", security = @SecurityRequirement(name = "bearerAuth"))
   public ResponseEntity<CourseDto> enrollCourse(@PathVariable long courseId) {
-    String username =
-        (String) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
 
+    String username = SecurityContextHolder.getContext().getAuthentication().getName();
     return userService
         .enrollCourse(username, courseId)
         .map(
             course ->
                 ResponseEntity.status(HttpStatus.CREATED).body(CourseMapper.toCourseDto(course)))
         .orElseGet(() -> ResponseEntity.notFound().build());
   }
+
+  @PutMapping("/{userId}/mentors/{mentorId}")
+  @Operation(summary = "save mentor", security = @SecurityRequirement(name = "bearerAuth"))
+  public ResponseEntity<UserWithSavedMentorsDto> saveMentor(
+      @PathVariable Long userId, @PathVariable Long mentorId) {
+
+    String username = SecurityContextHolder.getContext().getAuthentication().getName();
+    return ResponseEntity.ok(
+        UserMapper.userToUserWithSavedMentorsDto(
+            userService.saveMentor(userId, mentorId, username)));
+  }
+
+  @DeleteMapping("/{userId}/mentors/{mentorId}/saved")
+  @Operation(summary = "delete saved mentor", security = @SecurityRequirement(name = "bearerAuth"))
+  public ResponseEntity<Void> deleteMentor(@PathVariable Long userId, @PathVariable Long mentorId) {
+
+    String username = SecurityContextHolder.getContext().getAuthentication().getName();
```

Comment:
let's not pass logged-in username to the service method, the service doesn't need to know about the logged-in username to remove a mentor from the saved list, do this check before calling the service method. like some validation